3.processing_features/scripts/0.merge_sc_cytotable.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Merge single cells from CellProfiler outputs using CytoTable
+
+# ## Import libraries
+
+# In[1]:
+
+
+# Modified Jenna's nf1_ic.ipynb file from the Cellpainting repo
+# https://github.com/WayScience/nf1_cellpainting_data/blob/main/3.processing_features/0.merge_sc_cytotable.ipynb
+
+import sys
+import pathlib
+import yaml
+import pprint
+import pandas as pd
+
+# cytotable will merge objects from SQLite file into single cells and save as parquet file
+from cytotable import convert, presets
+
+# import utility to use function that will add single-cell count per well as a metadata column
+sys.path.append("../../utils/")
+import extraction_utils as sc_utils
+
+
+# ## Set paths and variables
+
+# In[2]:
+
+
+# type of file output from CytoTable (currently only parquet)
+dest_datatype = "parquet"
+
+# preset configurations based on typical CellProfiler outputs
+preset = "cellprofiler_sqlite_pycytominer"
+
+# update preset to include site metadata and cell counts
+joins = presets.config["cellprofiler_sqlite_pycytominer"]["CONFIG_JOINS"].replace(
+    "Image_Metadata_Well,",
+    "Image_Metadata_Well, Image_Metadata_Site,",
+)
+
+# set main output dir for all parquet files
+output_dir = pathlib.Path("../outputs/converted_data/")
+output_dir.mkdir(exist_ok=True)
+
+# directory where SQLite files are located
+sqlite_dir = pathlib.Path("../../2.cellprofiler_analysis/outputs/SQLites/")
+
+# list for pipeline names based on files to use to create dictionary
+pipeline_names = []
+
+# iterate through "IC_function_apply_pipelines" and append pipline_names from file names
+for file_path in pathlib.Path("../../2.cellprofiler_analysis/pipelines/IC_function_apply_pipelines").iterdir():
+    if str(file_path.stem).startswith("IC_function"):
+        pipeline_names.append(str(file_path.stem))
+
+print(pipeline_names)
+
+
+# ## Create dictionary with info for each plate
+# 
+# **Note:** All paths must be string to use with CytoTable.
+
+# In[3]:
+
+
+# create plate info dictionary with all parts of the CellProfiler CLI command to run in parallel
+plate_info_dictionary = {
+    name: {
+        "dest_path": str(pathlib.Path(f"{output_dir}/{name}.parquet")),
+    }
+    for name in pipeline_names 
+}
+
+# iterate over the dictionary and add the source_path
+for name, info in plate_info_dictionary.items():
+        for i in range(len(pipeline_names)):
+            if name == pipeline_names[i]:
+                  info["source_path"] = str(pathlib.Path(f"../../2.cellprofiler_analysis/outputs/SQLites/" + pipeline_names[i]+ ".sqlite"))
```

Comment:
please see above comment about fstrings :)

Reply:
Got it, thank you for showing me this!